elliptic-curve/src/osswu.rs

```diff
@@ -0,0 +1,84 @@
+use ff::Field;
+use subtle::Choice;
+
+/// The Optimized Simplified Shallue-van de Woestijne-Ulas parameters
+pub struct OsswuMapParams<F>
+where
+    F: Field,
+{
+    /// The first constant term
+    pub c1: [u64; 4],
+    /// The second constant term
+    pub c2: F,
+    /// The ISO A variable or Curve A variable
+    pub map_a: F,
+    /// The ISO A variable or Curve A variable
+    pub map_b: F,
+    /// The Z parameter
+    pub z: F,
+}
+
+/// Trait for determining the parity of the field
+pub trait Sgn0: Field {
+    /// Return the parity of the field
+    /// 1 == negative
+    /// 0 == non-negative
+    fn sgn0(&self) -> u8;
+}
+
+/// The optimized simplified Shallue-van de Woestijne-Ulas method
+/// for mapping elliptic curve scalars to affine points.
+pub trait OsswuMap: Sgn0 {
```

Comment:
I think it might be more straightforward if this were:

```suggestion
pub trait OsswuMap: Field + Sgn0 {
```

...since this trait is the one that actually uses methods of `Field`.

I think you could debatably remove the field bound from `Sgn0` in that case as well.